src/toil/batchSystems/mesos/batchSystem.py

```diff
@@ -352,8 +352,9 @@ def resourceOffers(self, driver, offers):
 
         # Right now, gives priority to largest jobs
         for offer in offers:
-            tasks = []
             # TODO: In an offer, can there ever be more than one resource with the same name?
+            allTasks = []
+            subTasks = defaultdict(list)
```

Comment:
Using the term sub-task is confusing here. A sub task is a task that needs to be finished, before its super task can be considered finished. This does not apply here. Rename to `tasksByType`.